pd/src/staking.rs

```diff
@@ -1,12 +1,32 @@
+use num::bigint;
+use num::rational;
+use penumbra_crypto::{Address, Value};
+use serde::{Deserialize, Serialize};
 use tendermint::{vote, PublicKey};
 
 const PENUMBRA_BECH32_VALIDATOR_PREFIX: &str = "penumbravalpub";
 /// Validator tracks the Penumbra validator's long-term consensus key (tm_pubkey), as well as their
 /// voting power.
-#[derive(Debug, Eq, PartialOrd, Ord, Clone, Copy)]
+#[derive(Deserialize, Serialize, Debug, Eq, Clone)]
 pub struct Validator {
+    /// tm_pubkey is the validator's long-term Tendermint public key, where the private component
+    /// is used to sign blocks.
     tm_pubkey: PublicKey,
+
+    /// voting_power is the validator's current voting power.
     pub voting_power: vote::Power,
+
+    /// commission_address is the validator's shielded commission address, where they receive their
+    /// portion of the staking rewards.
+    #[serde(with = "serde_with::rust::display_fromstr")]
+    pub commission_address: Address,
+
+    /// commission_rate is the portion of staking rewards that go to the validator (as opposed to
+    /// the delegators).
+    pub commission_rate: rational::Ratio<bigint::BigInt>,
```

Comment:
Rather than having an arbitrary-precision rational, what if we specified the commission rate as a fixed-precision integer? 1 basis point is probably enough precision, and 100% is 10,000 bps, so this could be
```rust
pub commission_rate_bps: u16
```
(though there's no particular reason to prefer a `u16` over a bigger integer other than that it fits the entire range)

I think this will make ensuring that we do correct rounding much easier later on.

Reply:
This sounds good, I implemented it this way due to (a) being very paranoid about future arithmetic rounding issues and (b) this seems to be how the Cosmos SDK does it (using Go's `big.Int`.) Arbitrary precision rationals are probably overkill here

Reply:
Yeah, that makes sense, and definitely better than using a float or something, but I think that the rounding behavior will be easier to analyze with fixed-point arithmetic.